backend/src/graphql/resolver/UserResolver.test.ts

```diff
@@ -668,7 +668,112 @@ describe('UserResolver', () => {
       })
 
       it('logs the error thrown', () => {
-        expect(logger.error).toBeCalledWith('The User has no valid credentials.')
+        expect(logger.error).toBeCalledWith('No user with this credentials', variables.email)
+      })
+    })
+
+    describe('user is in database but deleted', () => {
+      beforeAll(async () => {
+        jest.clearAllMocks()
+        await userFactory(testEnv, stephenHawking)
+        const variables = {
+          email: stephenHawking.email,
+          password: 'Aa12345_',
+          publisherId: 1234,
+        }
+        result = await mutate({ mutation: login, variables })
+      })
+
+      afterAll(async () => {
+        await cleanDB()
+      })
+
+      it('returns an error', () => {
+        expect(result).toEqual(
+          expect.objectContaining({
+            errors: [
+              new GraphQLError('This user was permanently deleted. Contact support for questions'),
+            ],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith(
+          'This user was permanently deleted. Contact support for questions',
+          expect.objectContaining({
+            firstName: stephenHawking.firstName,
+            lastName: stephenHawking.lastName,
+          }),
+        )
+      })
+    })
+
+    describe('user is in database but email not confirmed', () => {
+      beforeAll(async () => {
+        jest.clearAllMocks()
+        await userFactory(testEnv, garrickOllivander)
+        const variables = {
+          email: garrickOllivander.email,
+          password: 'Aa12345_',
+          publisherId: 1234,
+        }
+        result = await mutate({ mutation: login, variables })
+      })
+
+      afterAll(async () => {
+        await cleanDB()
+      })
+
+      it('returns an error', () => {
+        expect(result).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError('The Users email is not validate yet')],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith(
+          'The Users email is not validate yet',
+          expect.objectContaining({
+            firstName: garrickOllivander.firstName,
+            lastName: garrickOllivander.lastName,
+          }),
+        )
+      })
+    })
+
+    describe.skip('user is in database but password is not set', () => {
+      beforeAll(async () => {
+        jest.clearAllMocks()
+        // TODO: we need an user without password set
```

Comment:
This is Garrick Ollivander. So what is missing?

Reply:
This is not correct. Garrick Ollivander is an user with unconfirmed EMail and unset password.

See the test above. As mentioned in the resolver we cannot test this properly currently.

![image](https://user-images.githubusercontent.com/1238238/216202964-485b9d2f-3411-4f56-8ae4-6d6e06b3463a.png)